src/turret.cpp

```diff
@@ -589,16 +590,20 @@ int vehicle::automatic_fire_turret( vehicle_part &pt )
         if( auto_target == nullptr ) {
             if( boo_hoo ) {
                 cpu.name = string_format( pgettext( "vehicle turret", "The %s" ), pt.name() );
-                if( u_see ) {
+                // check if the player can see or hear then print chooses a message accordingly
+                if( u_see & u_hear ) {
                     add_msg( m_warning, ngettext( "%s points in your direction and emits an IFF warning beep.",
                                                   "%s points in your direction and emits %d annoyed sounding beeps.",
                                                   boo_hoo ),
                              cpu.name, boo_hoo );
-                } else {
+                } else if( !u_see & u_hear ) {
                     add_msg( m_warning, ngettext( "%s emits an IFF warning beep.",
                                                   "%s emits %d annoyed sounding beeps.",
                                                   boo_hoo ),
                              cpu.name, boo_hoo );
+                } else if( u_see & !u_hear ) {
+                    add_msg( m_warning, "%s points in your direction",
+                             cpu.name, boo_hoo );
```

Comment:
1. Missing period: `"%s points in your direction."`
2. There is only one `%s` but you provided `cpu.name` and `boo_hoo`. I guess only `cpu.name` is used here, is it right?

```suggestion
                    add_msg( m_warning, "%s points in your direction.", cpu.name );
```

Reply:
```suggestion
                    add_msg( m_warning, _( "%s points in your direction" ), cpu.name );
```
The string should also be translated.